desktop/Dockerfile

```diff
@@ -20,6 +20,11 @@ RUN apt update && apt install -y gnupg wget software-properties-common && \
       python3-pytest python3-mock xvfb qttools5-dev-tools pyqt5-dev-tools && \
     apt-get clean
 
+# Upstream script to run plugin tests (only for tags & master)
+RUN ([[ "$qgis_version" == "master" ]] && upstream_version="master" || { [[ "$qgis_version" =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]] && UPSTREAM="final-${VERSION//./_}"; }) && \
```

Comment:
tbh, I put this together very quickly and didn't _really_ go through the entire logic that determines `$qgis_version`. 

or maybe you'd rather like to port that script into this repo? probably makes more sense? it's not used upstream in any other code, but no idea what other workflows might depend on that script.

Reply:
I would rather move this logic in the workflow to determine a new variable `qgis_branch` which would be used as an arugment here.
also, in the current code it seems there is a mismatch between `upstream_version` and `UPSTREAM`